homeassistant/components/openweathermap/sensor.py

```diff
@@ -83,17 +86,19 @@ def __init__(
         sensor_type,
         sensor_configuration,
         weather_coordinator: WeatherUpdateCoordinator,
+        forecast_sensors_days,
     ):
         """Initialize the sensor."""
         super().__init__(
             name, unique_id, sensor_type, sensor_configuration, weather_coordinator
         )
         self._weather_coordinator = weather_coordinator
+        self._forecast_sensors_days = forecast_sensors_days
 
     @property
     def state(self):
         """Return the state of the device."""
         forecasts = self._weather_coordinator.data.get(ATTR_API_FORECAST)
         if forecasts is not None and len(forecasts) > 0:
-            return forecasts[0].get(self._sensor_type, None)
+            return forecasts[self._forecast_sensors_days].get(self._sensor_type, None)
```

Comment:
I'm having trouble understanding how this is intended to work. It only creates one sensor N days ahead of today? And to get more than one day you have to configure the integration more than once?

Why not always create N sensors with one forecast sensor for each day? The data already appears to be available in _weather_coordinator anyway. Would be an extra loop nested inside `for sensor_type in FORECAST_MONITORED_CONDITIONS:`.

Reply:
> It only creates one sensor N days ahead of today?
No, overall it creates set of sensors, for each value from `forecast[0]` dict.

Previously it takes only first day from `forecast` list (which is today), I added a property where you can define for which day of forecast set of sensors will be created.

So if you set value to 1, it will create forecast sensors which has forecast values for tomorrow.

Reply:
Create all sensors instead and disable them by default for the forecast day that is not today. Remove the config option.

Reply:
Are you sure? Then it will be created ~35 sensors

Reply:
Yes, please.